TreeMatching.java

```diff
@@ -0,0 +1,158 @@
+package Week6;
+import static org.testng.Assert.assertEquals;
+import org.junit.Test;
+
+/*
+Microsoft Interview Coaching Pilot 2018 (MICP 2018)
+Author: Aishna Gupta
+Language: JAVA
+
+************WEEK 6************
+
+THREE SUM
+
+1. TALK
+
+Given two non-empty binary trees s and t, check whether tree t has exactly the same structure and node values with a subtree of s. A subtree of s is a tree consists of a node in s and all of this node's descendants. 
+
+INPUT FORMAT:
+Root node of two trees
+
+OUTPUT FORMAT:
+Boolean
+
+SAMPLE INPUT: Root1 = 1, Root2 = 3 
+		 1
+		/ \
+ 	   2   3	, 	 3
+	  / \ / \		/ \
+	 4  5 6  7	   4   5
+SAMPLE OUTPUT:
+
+True
+
+*/
+
+/*
+
+SOLUTION
+	 
+	 MatchTree(Node root1, Node root2):
+	 	if(root1==null && root2==null):
+	 		return true;
+	 	
+	 	if(root1==null || root2==null)
+	 		return false;
+	 	
+	 	if(root1.data==root2.data && MatchTree(root1.left,root2.left) && MatchTree(root1.right,root2.right)):
+	 		return true;
+	 	
+	 	return MatchTree(root1.left,root2) || MatchTree(root1.right,root2);
+*/
+
+
+public class TreeMatching {
+
+	public boolean matchTree(TreeNode root1, TreeNode root2){
+		if(root1==null && root2==null){
+			return true;
+		}
+		
+		if(root1==null || root2==null){
+			return false;
+		}
+		if(root1.data==root2.data && matchTree(root1.left, root2.left) && matchTree(root1.right, root2.right)){
+			return true;
+		}
+		return matchTree(root1.left, root2) || matchTree(root1.right, root2);
+	}
+	
+//	TEST CASES
+	
+	@Test
+	public void NullTest(){
+		TreeNode root1 = null;
+		TreeNode root2 = null;	
+		assertEquals(true, matchTree(root1, root2));
```

Comment:
Another test case might be if only one the the trees is null instead of both

Reply:
A question I might ask the interviewer: if root 1 is nonempty and root2 is NULL, is the answer true?  I think this might be true in the same way the empty set is a subset of any set.  I don't think your solution covers this case.